ipaclient/plugins/vault.py

```diff
@@ -568,14 +580,119 @@ def forward(self, *args, **options):
         return response
 
 
+class _TransportCertInvalid(Exception):
+    def __init__(self, exc_info):
+        self.exc_info = exc_info
+
+
+_transport_cert_cache = {}
+
+
+class ModVaultData(Local):
+    def _do_internal(self, mechanism, session_key, transport_cert_der,
+                     *args, **options):
+        nss_transport_cert = nss.Certificate(transport_cert_der)
+
+        # wrap session key with transport certificate
+        # pylint: disable=no-member
+        public_key = nss_transport_cert.subject_public_key_info.public_key
+        # pylint: enable=no-member
+        wrapped_session_key = nss.pub_wrap_sym_key(mechanism,
+                                                   public_key,
+                                                   session_key)
+
+        options['session_key'] = wrapped_session_key.data
+
+        name = self.name + '_internal'
+        try:
+            return self.api.Command[name](*args, **options)
+        except errors.NotFound:
+            raise
+        except (errors.InternalError,
+                errors.ExecutionError,
+                errors.GenericError):
+            raise _TransportCertInvalid(sys.exc_info())
+
+    def internal(self, mechanism, session_key, *args, **options):
+        """
+        Calls the internal counterpart of the command.
+        """
+        domain = self.api.env.domain
+        dirname = TRANSPORT_CERT_CACHE_PATH
+        basename = DNSName(domain).ToASCII() + '.pem'
```

Comment:
Seems a bit redundant to me, since the full path is `~/.cache/ipa/kra-transport-certs/$DOMAIN.pem`.